### PR TITLE
update nightly version to 2024-09-01

### PR DESCRIPTION
**Pull request by Velfi**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/visitor.rs [hunk 1]
```diff
@@ -439,6 +438,7 @@ impl Visitor {
                             self.visit_generic_param_defs(path, generic_params)?;
                         }
                         GenericBound::Outlives(_) => {}
+                        GenericBound::Use(_) => {},
```
Comment (jdisanti) [inline src/visitor.rs:441]: What is this? Does it refer to any types we need to check?

Comment (robjtede) [reply src/visitor.rs:441]: It'll be related to the new precise capture syntax. See https://rust-lang.github.io/rfcs/3617-precise-capturing.html?highlight=use

**Review by jdisanti — APPROVED** [review]
Looks good, thank you! Some minor CI fixes need to be done for merge.

**Comment by Velfi** [comment]
Closing in favor of https://github.com/awslabs/cargo-check-external-types/pull/190

**Pull request closed without merging**